lib/google_crawler_web/controllers/keyword_controller.ex

```diff
@@ -14,7 +14,16 @@ defmodule GoogleCrawlerWeb.KeywordController do
   end
 
   def show(conn, %{"id" => id}) do
-    keyword = Repo.get_by(Keyword, %{id: id})
-    render(conn, "show.html", keyword: keyword)
+    keyword = Repo.get_by(Keyword, %{id: id, user_id: conn.assigns[:current_user].id})
```

Comment:
Another way is to use `get_by!`, so it raises `Ecto.NoResultsError` Then it is going to be catch by phoenix and render 404 error page 💡